lib/runner.js

```diff
@@ -574,13 +574,28 @@ internals.protect = function (item, state) {
         /* $lab:coverage:off$ */
         const promiseRejectionHandler = function (err) {
 
+            if (flags.onUnhandledRejection) {
+                flags.onUnhandledRejection(err);
```

Comment:
If this throws - it's handled by `onUncaughtException` - I think `try/catch` is not necessary here?